app/controllers/publish/courses/fee_or_salary_controller.rb

```diff
@@ -19,6 +19,26 @@ def continue
         end
       end
 
+      def update
+        authorize(provider)
```

Comment:
Think this should be tighter, as only course that is not published can be edited right?

Reply:
agreed, done